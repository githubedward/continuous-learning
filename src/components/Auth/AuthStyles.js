import styled from "styled-components";
import { appear } from "../shared/style-variables";

// styles
export const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 3.5rem 3rem;
  background: #ffffff95;
  border-radius: 3px;
  animation: ${appear} 500ms ease-in;
`;
export const Title = styled.h1`
  margin-bottom: 1rem;
`;
export const Label = styled.label`
  margin: 0.5rem;
`;
export const Input = styled.input`
  padding: 0.2rem;
  font-size: 1.25rem;
`;
export const Button = styled.button`
  margin-top: 3rem;
  color: #ffffff;
  font-size: 1rem;
  width: 7.5rem;
  padding: 0.5rem 1rem;
  background: #fb720e;
  border: 1px solid #fb720e;
  border-radius: 3px;
  text-decoration: none;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  &:hover {
    background: #fd8912;
    transition: 0.3s;
  }
`;
export const Span = styled.span`
  color: #fd8912;
`;
export const Error = styled.span`
  color: red;
  margin-top: 0.5rem;
  font-weight: 300;
`;
