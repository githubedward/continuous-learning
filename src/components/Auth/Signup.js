import React from "react";
import PropTypes from "prop-types";
import {
  H1,
  Form,
  Label,
  Input,
  Button,
  Span,
  Footer,
  Anchor,
  Status
} from "./AuthStyles";
import PageContainer from "../shared/PageContainer";
import BrandName from "../shared/BrandName";
import * as styleGuides from "../shared/style-variables";

const Signup = ({
  handleUsername,
  handlePassword,
  handleFullName,
  handleSignup,
  updateStatus,
  linkLogin,
  fullname,
  username,
  password,
  status
}) => {
  const userInfo = {
    fullname,
    username,
    password
  };
  return (
    <PageContainer>
      <BrandName />
      <H1>
        <Span color={styleGuides.gray}>
          <strong>Signup</strong> to find and share beautiful experiences
        </Span>
      </H1>
      <Form
        onSubmit={e => {
          e.preventDefault();
          handleSignup(userInfo);
        }}
      >
        <Label>
          <Input
            type="text"
            name="full-name"
            placeholder="Full Name"
            value={fullname}
            onChange={e => {
              handleFullName(e.target.value);
            }}
          />
        </Label>
        <Label>
          <Input
            type="text"
            name="username"
            placeholder="Username"
            value={username}
            onChange={e => {
              handleUsername(e.target.value);
            }}
          />
        </Label>
        <Label>
          <Input
            type="password"
            name="password"
            placeholder="Password"
            value={password}
            onChange={e => {
              handlePassword(e.target.value);
            }}
          />
        </Label>
        <Status>{status}</Status>
        <Button>Signup</Button>
      </Form>
      <Footer>
        <Span fontWeight="lighter">Have an account?</Span>{" "}
        <Anchor onClick={linkLogin}>Login</Anchor>
      </Footer>
    </PageContainer>
  );
};

Signup.propTypes = {
  handleUsername: PropTypes.func.isRequired,
  handlePassword: PropTypes.func.isRequired,
  handleFullName: PropTypes.func.isRequired,
  handleSignup: PropTypes.func.isRequired,
  updateStatus: PropTypes.func.isRequired,
  fullname: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
};

Signup.defaultProps = {
  handleUsername: () => alert("default"),
  handlePassword: () => alert("default"),
  handleFullName: () => alert("default"),
  handleSignup: () => alert("default"),
  updateStatus: () => alert("default"),
  fullname: "default",
  username: "default",
  password: "default",
  status: "default"
};

export default Signup;
