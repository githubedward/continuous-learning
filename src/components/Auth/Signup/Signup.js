import React from "react";
import PropTypes from "prop-types";
import PageContainer from "../../shared/PageContainer";
import BrandName from "../../shared/BrandName";
import * as styleGuides from "../../shared/style-variables";
import { MagicSpinner } from "react-spinners-kit";
import { H1, Span, Footer, Status } from "../AuthStyles";
import Anchor from "../../shared/Anchor";
import Button from "../../shared/Button";
import { Form, Input, input } from "../../shared/FormComponents";
import { ThemeProvider } from "styled-components";
import { btnTheme } from "../AuthStyles";

const Signup = ({
  handleUsername,
  handlePassword,
  handleFullName,
  handleSignup,
  isLoading,
  linkLogin,
  fullname,
  username,
  password,
  status
}) => {
  const userInfo = {
    fullname,
    username,
    password
  };

  return (
    <PageContainer>
      <BrandName size={"3.5rem"} />
      <H1>
        <Span color={styleGuides.gray}>
          <Span fontWeight="bold" color={styleGuides.gray}>
            Signup
          </Span>{" "}
          to find and share beautiful experiences
        </Span>
      </H1>
      <Form
        onSubmit={e => {
          e.preventDefault();
          handleSignup(userInfo);
        }}
        padding="2rem 2.5rem"
      >
        {input("text", "fullname", "Full Name", fullname, e => {
          handleFullName(e.target.value);
        })}
        {input("text", "username", "Username", username, e => {
          handleUsername(e.target.value);
        })}
        {input("password", "password", "Password", password, e => {
          handlePassword(e.target.value);
        })}
        <Status>{status}</Status>
        <ThemeProvider theme={btnTheme}>
          <Button>
            {isLoading ? (
              <MagicSpinner loading={isLoading} size={25} color={"white"} />
            ) : (
              "Signup"
            )}
          </Button>
        </ThemeProvider>
      </Form>
      <Footer>
        <Span fontWeight={styleGuides.light}>Have an account?</Span>{" "}
        <Anchor onClick={linkLogin}>Login</Anchor>
      </Footer>
    </PageContainer>
  );
};

Signup.propTypes = {
  handleUsername: PropTypes.func.isRequired,
  handlePassword: PropTypes.func.isRequired,
  handleFullName: PropTypes.func.isRequired,
  handleSignup: PropTypes.func.isRequired,
  updateStatus: PropTypes.func.isRequired,
  fullname: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
};

Signup.defaultProps = {
  handleUsername: () => alert("default"),
  handlePassword: () => alert("default"),
  handleFullName: () => alert("default"),
  handleSignup: () => alert("default"),
  updateStatus: () => alert("default"),
  fullname: "default",
  username: "default",
  password: "default",
  status: "default"
};

export default Signup;
