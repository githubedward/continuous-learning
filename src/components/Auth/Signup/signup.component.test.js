import Signup from "./Signup";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import { MagicSpinner } from "react-spinners-kit";
import { Status } from "../AuthStyles";

Enzyme.configure({ adapter: new Adapter() });

describe("<Signup />", () => {
  const props = {
    handleUsername: jest.fn(),
    handlePassword: jest.fn(),
    handleFullName: jest.fn(),
    handleSignup: jest.fn(),
    linkSignup: jest.fn(),
    isLoading: false,
    username: "username",
    password: "password",
    status: "status"
  };

  // shallow rendering
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Signup {...props} />);
  });

  it("should render without errors", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // Full DOM rendering
  let component;
  beforeEach(() => {
    component = mount(<Signup {...props} />);
  });

  it("should render button correctly", () => {
    const button = component.find("button");
    expect(button.text()).toBe("Signup");
    props.isLoading = true;
    expect(button.find(MagicSpinner)).toBeTruthy();
  });

  it("should handle fullname input change", () => {
    const input = component
      .find("input")
      .filterWhere(el => el.prop("name") === "fullname");
    expect(input.prop("type")).toBe("text");
    expect(input.prop("placeholder")).toBe("Full Name");
    input.simulate("change");
    expect(component.props().handleFullName).toHaveBeenCalled();
  });

  it("should handle username input change", () => {
    const input = component
      .find("input")
      .filterWhere(el => el.prop("name") === "username");
    expect(input.prop("type")).toBe("text");
    expect(input.prop("placeholder")).toBe("Username");
    input.simulate("change");
    expect(component.props().handleUsername).toHaveBeenCalled();
  });

  it("should handle password input change", () => {
    const input = component
      .find("input")
      .filterWhere(el => el.prop("name") === "password");
    expect(input.prop("type")).toBe("password");
    expect(input.prop("placeholder")).toBe("Password");
    input.simulate("change");
    expect(component.props().handlePassword).toHaveBeenCalled();
  });

  it("should call fn when form is submitted", () => {
    const form = component.find("form");
    form.simulate("submit");
    expect(props.handleSignup).toHaveBeenCalled();
  });

  it("should render link", () => {
    expect(component.find("a")).toBeTruthy();
  });

  it("should update status", () => {
    const status = component.find(Status);
    expect(status.text()).toBe("status");
    component.setProps({ status: "test" });
    expect(status.text()).toBe("test");
  });
});
