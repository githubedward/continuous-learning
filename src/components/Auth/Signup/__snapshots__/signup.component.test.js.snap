// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should match the snapshot 1`] = `
<styled.section>
  <BrandName
    size="3.5rem"
  />
  <styled.h1>
    <styled.span
      color="#B4B4B4"
    >
      <styled.span
        fontWeight="bold"
      >
        Signup
      </styled.span>
       to find and share beautiful experiences
    </styled.span>
  </styled.h1>
  <styled.form
    onSubmit={[Function]}
  >
    <styled.input
      name="fullname"
      onChange={[Function]}
      placeholder="Full Name"
      type="text"
      value="default"
    />
    <styled.input
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value="username"
    />
    <styled.input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="password"
    />
    <styled.span>
      status
    </styled.span>
    <ThemeProvider
      theme={
        Object {
          "bgc": "#ed4949",
          "borderColor": "#ed4949",
          "color": "white",
          "fontSize": "1.25rem",
          "fweight": 500,
          "hoverbgc": "#ff3030",
          "margin": "3rem 0 0 0",
          "padding": ".5rem 1rem",
          "width": "inherit",
        }
      }
    >
      <styled.button>
        Signup
      </styled.button>
    </ThemeProvider>
  </styled.form>
  <styled.footer>
    <styled.span
      fontWeight={300}
    >
      Have an account?
    </styled.span>
     
    <styled.a>
      Login
    </styled.a>
  </styled.footer>
</styled.section>
`;
